### PR TITLE
Support `pad_with` argument in `AmplitudeEmbedding` for all interfaces

### PR DESCRIPTION
**Pull request by eddddddy**
**Context:**
The `AmplitudeEmbedding` template has an optional argument `pad_with` that pads the input features if the shape is smaller than the dimension of the feature space. Passing a value for this argument is incompatible with some interfaces if the input features  is an interface-specific tensor.

**Description of the Change:**
Allow passing in `pad_with` argument even if `features` is a tensor. Add tests for `pad_with` for all interfaces.

**Related GitHub Issues:**
https://github.com/PennyLaneAI/pennylane/issues/3379

### CODE REVIEW

File: pennylane/templates/embeddings/amplitude.py [hunk 1]
```diff
@@ -211,13 +211,7 @@ def _preprocess(features, wires, pad_with, normalize):
...
                 if n_features < dim:
                     padding = [pad_with] * (dim - n_features)
-                    if (
-                        hasattr(feature_set, "device") and feature_set.device.type == "cuda"
-                    ):  # pragma: no cover
-                        ## Torch tensor, send to same GPU
-                        dat_type = type(feature_set)
-                        padding = dat_type(padding).to(feature_set.device)
-
+                    padding = qml.math.convert_like(padding, feature_set)
```
Comment (rmoyard) [inline pennylane/templates/embeddings/amplitude.py:214]: Does it also work with GPU with Torch? I added the GPU tests

File: tests/templates/test_embeddings/test_amplitude.py [hunk 2]
```diff
@@ -487,3 +544,32 @@ def test_torch(self, tol, features):
         res2 = circuit2(features)
 
         assert qml.math.allclose(res, res2, atol=tol, rtol=0)
+
+    @pytest.mark.torch
+    @pytest.mark.parametrize("features", all_features)
+    def test_autograd_pad_with(self, tol, features):
```
Comment (rmoyard) [inline tests/templates/test_embeddings/test_amplitude.py:550]: It should be torch here.

File: tests/templates/test_embeddings/test_amplitude.py [hunk 3]
```diff
@@ -487,3 +544,32 @@ def test_torch(self, tol, features):
...
+
+        @qml.qnode(dev, interface="torch")
+        def circuit(x=None):
+            qml.AmplitudeEmbedding(x, list(range(4)), pad_with=0.0)
+            return qml.state()
+
+        res = circuit(features)
+        res2 = circuit_decomposed(features)
+
+        assert qml.math.allclose(res, res2, atol=tol, rtol=0)
```
Comment (rmoyard) [inline tests/templates/test_embeddings/test_amplitude.py:575]: It would be great to add a Jax test / Jax jit test

**Review by rmoyard — COMMENTED** [review]
Thanks @eddddddy, it looks good! Just some worries about torch and GPU. I added the test for the PR.

File: tests/templates/test_embeddings/test_amplitude.py (hunk 3 above)
Comment (eddddddy) [reply tests/templates/test_embeddings/test_amplitude.py:575]: The JAX-JIT already exists. It was added here: https://github.com/PennyLaneAI/pennylane/pull/3166

File: pennylane/templates/embeddings/amplitude.py (hunk 1 above)
Comment (eddddddy) [reply pennylane/templates/embeddings/amplitude.py:214]: Yes, `qml.math.convert_like` already moves torch tensors to the correct device.

File: tests/templates/test_embeddings/test_amplitude.py (hunk 2 above)
Comment (eddddddy) [reply tests/templates/test_embeddings/test_amplitude.py]: Fixed.

**Review by rmoyard — APPROVED** [review]
Just add the changelog and you can merge it 💯

**Pull request merged**